From f234b577fb3167c3b9d6f7a14fa7f8903454cdd0 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 08 Feb 2023 16:30:09 +0200
Subject: [PATCH] [SW-120178] app: introduce app specific defs

Introduce a way (and a place) for app specific definitions.
Replace adhoc ones by the new way.

Change-Id: I404755aa662f47e6f50805d4a01dc8b66e7c12ac
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Makefile b/Makefile
index a74d03a..f5cb9eb 100644
--- a/Makefile
+++ b/Makefile
@@ -20,10 +20,10 @@
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV)
 CFLAGS += -I ${EMBEDDED_ROOT}/arcboot/arch
 CFLAGS += -include $(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV)/$(SPECS_EMB_DEV)_reg_map.h
+CFLAGS += -include app/preboot/preboot_app_defs.h
 CFLAGS += -Wall -Werror
 CFLAGS += -g -c -fno-builtin
 CFLAGS += -I ./include/ -I ./ -I ./lib/ -I ./lib/common/ -I ./$(HL_DEV_ARCH)/ -I ./$(HL_DEV_ARCH_COMMON)/
-CFLAGS += -D DEF_HL_FW_COMP_PREBOOT
 
 ifeq ($(HL_TARGET),pldm)
 	CFLAGS += -DTARGET_IS_PLDM
diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index 922db16..2a2335f 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -16,7 +16,7 @@
 CFLAGS += -I $(HL_DEV_ARCH)/ -I $(HL_DEV_ARCH)/../common/ -I $(SPECS_EMBEDDED_DIR)/$(BASE_DEV_NAME)/
 CFLAGS += -include $(SPECS_EMBEDDED_DIR)/$(SPECS_DEV)/$(SPECS_DEV)_reg_map.h
 CFLAGS += -include $(HL_DEV_ARCH)/dev_ext_rom.h
-CFLAGS += -D DEF_HL_FW_COMP_EROM
+CFLAGS += -include erom_app_defs.h
 CFLAGS += -I ${EMBEDDED_ROOT}/arcboot/arch
 # dummy parameter, real one required only for ppboot build
 CFLAGS += -DPHY_FW_SIZE=0
diff --git a/app/ext_rom/erom_app_defs.h b/app/ext_rom/erom_app_defs.h
new file mode 100644
index 0000000..67be318
--- /dev/null
+++ b/app/ext_rom/erom_app_defs.h
@@ -0,0 +1,11 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2023 HabanaLabs Ltd.
+ */
+
+#ifndef _EROM_APP_DEFS_H_
+#define _EROM_APP_DEFS_H_
+
+#define HL_FW_COMP	HL_FW_COMP_EROM
+
+#endif /* _EROM_APP_DEFS_H_ */
diff --git a/app/preboot/preboot_app_defs.h b/app/preboot/preboot_app_defs.h
new file mode 100644
index 0000000..e6c2e27
--- /dev/null
+++ b/app/preboot/preboot_app_defs.h
@@ -0,0 +1,11 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2023 HabanaLabs Ltd.
+ */
+
+#ifndef _PREBOOT_APP_DEFS_H_
+#define _PREBOOT_APP_DEFS_H_
+
+#define HL_FW_COMP	HL_FW_COMP_PREBOOT
+
+#endif /* _PREBOOT_APP_DEFS_H_ */
diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index 2ac3b8f..14730fc 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -117,13 +117,7 @@
 {
 	status->version = HL_UPD_PROTO_VERSION;
 	status->compat_mask = cpu_boot_dev_sts_get_all();
-#if defined(DEF_HL_FW_COMP_EROM)
-	status->fw_comp = HL_FW_COMP_EROM;
-#elif defined(DEF_HL_FW_COMP_PREBOOT)
-	status->fw_comp = HL_FW_COMP_PREBOOT;
-#else
-	status->fw_comp = HL_FW_COMP_NONE;
-#endif
+	status->fw_comp = HL_FW_COMP;
 	status->err = err;
 	status->sub_err = serr;
 }
